File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        stack<string> st;
        string str = "";
        for (int i = 0; i < s.size(); i++) {
            if (s[i] == ' ') {
                if (!str.empty()) {
                    st.push(str);
                    str = "";
                }
            } else
                str += s[i];
        }
        if (!str.empty())
            st.push(str);
        string ans = "";
        while (st.size() != 1) {
            ans += st.top() + " ";
            st.pop();
        }
        ans += st.top();
        return ans;
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        Stack<String> st = new Stack<>();
        int i = 0;
        int n = s.length();
        String temp = new String();
        while (i < n) {
            if (s.charAt(i) == ' ') {
                if (temp.length() != 0) {
                    st.push(temp);
                    temp = "";
                }
            } else if (s.charAt(i) != ' ') {
                temp += s.charAt(i);
            }
            i++;
        }
        if (temp.length() != 0) {
            st.push(temp);
        }
        String ans = "";
        while (!st.isEmpty()) {
            String tempS = st.pop();
            if (st.size() == 0) {
                ans += tempS;
            } else
                ans += tempS+" ";
        }

        return ans;
    }
}